tests/test-books-api.js

```diff
@@ -0,0 +1,178 @@
+            assert.strictEqual(response.data.State, State.SUCCESS)
+
+            let result = await dbbooks.get_book_by_id_async(response.data.createdBookId)
+            assert.strictEqual(result.rows[0].isbn, isbn)
+            assert.strictEqual(result.rows[0].name, name)
+            assert.strictEqual(result.rows[0].author, author)
+            assert.strictEqual(result.rows[0].publisher, publisher)
+            assert.strictEqual(result.rows[0].quantity, quantity)
+            assert.strictEqual(result.rows[0].id_category, id_category)
+        })
+
+        it('6.should return status 500 and the State response will be error when called with no parameters', async() => {
+            let response = await axios.post(`${process.env.API_URL}/books`, {}).catch(error => {
+                assert.strictEqual(error.response.status, 500)
+                assert.strictEqual(error.response.data.State, State.ERROR)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
+            })
+        })
+    })
+
+    describe('update Book', async() => {
+        it('7.should return status 200 and the data in database is the same with the data response when called', async() => {
+            const book_isbn = 'jlasdjf9381234'
+            const book_name = 'Naruto'
+            const book_author = 'Masashi Kishimoto'
+            const book_publisher = 'Kim Dong'
+            const book_quantity = 230
+            const book_id_category = '1c2ad358-98b0-4210-9ea3-3c4625b76e91'
+            let response = await axios.put(`${process.env.API_URL}/books/${book_id}`, {
+                isbn: book_isbn,
+                name: book_name,
+                author: book_author,
+                publisher: book_publisher,
+                quantity: book_quantity,
+                id_category: book_id_category
+            })
+            assert.strictEqual(response.status, 200)
+            assert.strictEqual(response.data.State, State.SUCCESS)
+
+            let result = await dbbooks.get_book_by_id_async(book_id)
+            assert.strictEqual(result.rows[0].isbn, book_isbn)
+            assert.strictEqual(result.rows[0].name, book_name)
+            assert.strictEqual(result.rows[0].author, book_author)
+            assert.strictEqual(result.rows[0].publisher, book_publisher)
+            assert.strictEqual(result.rows[0].quantity, book_quantity)
+            assert.strictEqual(result.rows[0].id_category, book_id_category)
+        })
+
+        it('8.should return status 500 and the State response will be error when called with no parameters', async() => {
+            let response = await axios.put(`${process.env.API_URL}/books/${book_id}`, {}).catch(error => {
+                assert.strictEqual(error.response.status, 500)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
+                assert.strictEqual(error.response.data.State, State.ERROR)
+            })
+        })
```

Comment:
Missing incorrect bookId and incorrect categoryId cases

Reply:
I fixed it